### PR TITLE
Implement basic testing scaffolding and GitHub Action

### PR DESCRIPTION
**Pull request by sirhcm**
Part of #7, some first steps in implementing tests for this repo. 

### **IMPORTANT: `MONGO_PASSWORD` in `.env` MUST be updated to `MONGO_URI`**

### CODE REVIEW

**Comment by colinpcurtis** [comment]
this looks good to me so far, but I think we should probably merge the graceful failures PR first just since this has breaking changes for that (tests would have to be updated).  Either way this is solid work and will be really pro level stuff.

File: commands/iam.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ const iam = async (email, user_id) => {
         user_id: user_id,
     }
 
-    await verification.insertOne(doc);
+    await db.collection("verification").insertOne(doc);
```
Comment (colinpcurtis) [inline commands/iam.js]: nit but these should be single quote to match rest of code

File: commands/update_roles.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-const { users } = require('./db')
+const db = require('../db')
 const {updateUserRoles} = require('./helpers');
 
 const updateRoles = async (user_id, client) => {
-    const data = await users.findOne({user_id});
+    const data = await db.collection("users").findOne({user_id});
```
Comment (colinpcurtis) [inline commands/update_roles.js]: same as above

File: commands/verify.js [hunk 3]
```diff
@@ -1,21 +1,21 @@
-const { verification, users } = require('./db')
+const db = require('../db')
 const { updateUserRoles } = require('./helpers')
 
 const addVerifiedUser = async (user_id, email, team) => {
     const doc = { user_id, email, team };
-    await users.insertOne(doc);
+    await db.collection("users").insertOne(doc);
```
Comment (colinpcurtis) [inline commands/verify.js:8]: here as well

File: commands/verify.js [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
 }
 
 const verify = async (code, user_id, client) => {
     const doc = {user_id, verification_code: code};
-    const userVerification = await verification.findOne(doc);
+    const userVerification = await db.collection("verification").findOne(doc);
```
Comment (colinpcurtis) [inline commands/verify.js]: and here

File: commands/verify.js [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
     if (!userVerification || userVerification.verification_code !== code) {
         return 'incorrect code';
     }
     const email = userVerification.email;
     const role = await updateUserRoles(email, user_id, client);
     addVerifiedUser(user_id, email, role);
-    await verification.deleteOne(doc);
+    await db.collection("verification").deleteOne(doc);
```
Comment (colinpcurtis) [inline commands/verify.js]: oop

File: commands/whoami.js [hunk 6]
```diff
@@ -1,9 +1,7 @@
-const { getDb } = require('./db')
+const db = require('../db')
 
 const whoami = async (user_id) => {
-    const db = getDb('users');
-    const user_info = await db.findOne({user_id});
-    return user_info;
+    return await db.collection("users").findOne({user_id});
```
Comment (colinpcurtis) [inline commands/whoami.js]: nice

File: db.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const { MongoClient } = require("mongodb");
+
+const connectionString = process.env.MONGO_URI;
+const client = new MongoClient(connectionString);
+
+const db = client.db("datares-discord-bot");
```
Comment (colinpcurtis) [inline db.js:6]: here as well

File: test/iam.test.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
+const { MongoClient } = require('mongodb');
+const mockSES = require("aws-sdk/clients/ses");
+const iam = require("../commands/iam");
+
```
Comment (colinpcurtis) [inline test/iam.test.js:4]: this is probably the best stuff in the PR, mocking SES and mongo is really pro level testing and will help a lot for development

**Review by colinpcurtis — CHANGES_REQUESTED** [review]

**Comment by sirhcm** [comment]
The double quotes is a Prettier thing, happy to change it, but if we end up merging #11 then it's kinda moot.

**Comment by colinpcurtis** [comment]
no more ping command #sad

**Review by colinpcurtis — APPROVED** [review]
looks good to me.  Testing is super great and this will really help the repo in the long term

**Pull request merged**